File: docs/examples/python/output/parts/UserManagementService_delete_user.py
# code2map fragment (non-buildable)
# original: docs/examples/python/user_management_service.py
# lines: 122-137
# symbol: UserManagementService#delete_user
# notes: references dataclasses.dataclass, re, typing.Optional; calls self._find_user_or_throw
    def delete_user(self, user_id: str) -> User:
        """
        ユーザーを削除する

        Args:
            user_id: 削除対象のユーザーID

        Returns:
            削除されたユーザー

        Raises:
            UserNotFoundException: ユーザーが見つからない場合
        """
        existing_user = self._find_user_or_throw(user_id)
        del self._users[user_id]
        return existing_user

File: code2map/docs/examples/v0.1.2/java/output/parts/UserManagementService_deleteUser.java
// code2map fragment (non-buildable)
// id: CD4
// original: docs/examples/java/UserManagementService.java
// lines: 98-102
// symbol: UserManagementService#deleteUser
// notes: references java.util.ArrayList, java.util.HashMap, java.util.List, java.util.Map, java.util.Optional, java.util.regex.Pattern; calls findUserOrThrow, users.remove
    public User deleteUser(String userId) {
        User existingUser = findUserOrThrow(userId);
        users.remove(userId);
        return existingUser;
    }
